web/test/pages/profile-foreign.test.tsx

```diff
@@ -0,0 +1,489 @@
+import { markdownToText } from "@/test/helpers/helpers-utilities";
+import * as mockRouter from "@/test/mock/mockRouter";
+import { useMockRouter } from "@/test/mock/mockRouter";
+import { WithStatefulProfileFormContext } from "@/test/mock/withStatefulProfileData";
+import { setupStatefulUserDataContext, WithStatefulUserData } from "@/test/mock/withStatefulUserData";
+import {
+  defaultDateFormat,
+  emptyIndustrySpecificData,
+  FormationData,
+  generateMunicipality,
+  generateProfileData,
+  getCurrentDateFormatted,
+  LookupLegalStructureById,
+  Municipality,
+  OperatingPhaseId,
+  ProfileData,
+  TaxFilingData,
+  UserData,
+} from "@businessnjgovnavigator/shared";
+import {
+  generateFormationData,
+  generateTaxFilingData,
+  generateUserData as _generateUserData,
+  randomLegalStructure,
+} from "@businessnjgovnavigator/shared/test";
+
+import { useMockRoadmap } from "@/test/mock/mockUseRoadmap";
+import { industryIdsWithRequiredEssentialQuestion } from "@/test/pages/onboarding/helpers-onboarding";
+import {
+  clickBack,
+  clickSave,
+  expectLocationNotSavedAndError,
+  expectLocationSavedAsUndefined,
+  removeLocationAndSave,
+} from "@/test/pages/profile-helpers";
+import { createTheme, ThemeProvider } from "@mui/material";
+import { fireEvent, render, screen, waitFor, within } from "@testing-library/react";
+
+const Config = getMergedConfig();
+
+jest.mock("next/router", () => ({ useRouter: jest.fn() }));
+jest.mock("@/lib/data-hooks/useUserData", () => ({ useUserData: jest.fn() }));
+jest.mock("@/lib/api-client/apiClient", () => ({ postGetAnnualFilings: jest.fn() }));
+jest.mock("@/lib/data-hooks/useRoadmap", () => ({ useRoadmap: jest.fn() }));
+
+const generateUserData = (overrides: Partial<UserData>): UserData => {
+  const profileData = generateProfileData({ ...overrides.profileData });
+  return _generateUserData({ ...overrides, profileData });
+};
+
+describe("profile-foreign", () => {
+  beforeEach(() => {
+    jest.resetAllMocks();
+    useMockRouter({});
+    useMockRoadmap({});
+    setupStatefulUserDataContext();
+  });
+
+  const renderPage = ({
+    municipalities,
+    userData,
+  }: {
+    municipalities?: Municipality[];
+    userData?: UserData;
+  }): void => {
+    const genericTown =
+      userData && userData.profileData.municipality
+        ? userData.profileData.municipality
+        : generateMunicipality({ displayName: "GenericTown" });
+    render(
+      withAuthAlert(
+        <ThemeProvider theme={createTheme()}>
+          <WithStatefulProfileFormContext>
+            <WithStatefulUserData initialUserData={userData}>
+              <Profile municipalities={municipalities ? [genericTown, ...municipalities] : [genericTown]} />
+            </WithStatefulUserData>
+          </WithStatefulProfileFormContext>
+        </ThemeProvider>,
+        IsAuthenticated.TRUE,
+        { registrationModalIsVisible: false }
+      )
+    );
+  };
```

Comment:
This render function looks pretty similar across all these tests. Is this something we could add to the helpers?